Adds feature CleanupJob for removing non-existent files (#10960)
JabRef/jabref: CHANGELOG.md

- We added the possibility to redownload files that had been present but are no longer in the specified location. [#10848](https://github.com/JabRef/jabref/issues/10848)
- We added the citation key pattern `[camelN]`. Equivalent to the first N words of the `[camel]` pattern.
- We added ability to export in CFF (Citation File Format) [#10661](https://github.com/JabRef/jabref/issues/10661).

> - We added CleanupJob for removing non-existent files and added a checkbox for it in Cleanup entries, as well as grouped the checkboxes in Cleanup entries. [#10929] (<https://github.com/JabRef/jabref/issues/10929>)

Reviewer: ```suggestion
- We added a Cleanup for removing non-existent files and grouped the related options [#10929](https://github.com/JabRef/jabref/issues/10929)
```
Context: Changelog is intended for end users, so no technical details
And no space between the link text in round braces and the link in square brackets

Author: Had misunderstood that there shouldn't be any technicality in changelog. It should be fixed now! Also updated the link so that it matches your suggestion!
